base/shell/explorer/syspager.cpp

```diff
@@ -178,6 +178,90 @@ class CNotifyToolbar :
     void Initialize(HWND hWndParent, CBalloonQueue * queue);
 };
 
+
+class CHardErrorThread
+{
+    DWORD m_ThreadId;
+    LONG m_bThreadRunning;
+    DWORD m_Status;
+    DWORD m_dwResponseType;
+    CStringW m_Title;
+    CStringW m_Text;
+public:
+
+    CHardErrorThread():
+        m_bThreadRunning(FALSE),
+        m_Status(NULL),
+        m_dwResponseType(NULL),
+        m_ThreadId(NULL)
+    {
+    }
+
+    ~CHardErrorThread()
+    {
+        if (m_bThreadRunning)
+        {
+            /* Try to unstuck Show */
+            PostThreadMessage(m_ThreadId, WM_QUIT, 0, 0);   
+
+            HANDLE hThread = OpenThread(THREAD_ALL_ACCESS, FALSE, m_ThreadId);
```

Comment:
If you need the handle again later, why close it in the first place? Could store it in the class instead of storing m_ThreadId.

Reply:
I thought that we don't need to keep the handle open for this rare case.

Reply:
It saves having to open the handle again in the rare case, where this can only lead to another obscure failure path.